core/src/cost_update_service.rs

```diff
@@ -174,6 +190,36 @@ impl CostUpdateService {
         );
         dirty
     }
+
+    // Based historical data, it seems reasonable to flag it as large change if updated_cost
+    // is more than double of input_data;
+    // Compare updated_cost against input_cost instead of previous calculated cost captures
+    // how far apart between single data point and calculated value, which is a good indicator
+    // of potential issues.
+    fn is_large_change_in_cost_calculation(input_cost: i64, updated_cost: i64) -> bool {
+        (updated_cost - input_cost).abs() > input_cost
+    }
+
+    fn report_large_change_in_cost(
+        program_id: &Pubkey,
+        program_timing: &ProgramTiming,
+        calculated_units: u64,
+    ) {
+        datapoint_info!(
+            "large_change_in_cost",
+            ("pubkey", program_id.to_string(), String),
+            ("execute_us", program_timing.accumulated_us, i64),
+            ("accumulated_units", program_timing.accumulated_units, i64),
+            ("count", program_timing.count, i64),
+            ("errored_units", program_timing.total_errored_units, i64),
+            (
+                "errored_count",
+                program_timing.errored_txs_compute_consumed.len(),
+                i64
+            ),
+            ("calculated_units", calculated_units as i64, i64),
+        );
```

Comment:
Reporting this for potentially every program deployed on every slot could be spammy. I think this should be throttled to a couple popular programs a slot.

Reply:
It only reports whe `is_large_change_in_cost_calculation` is true. Can set limit to be 95% of MAX_UNITS (eg 1_400_000) to return true?